pkg/route/api/validation/validation.go

```diff
@@ -111,8 +113,26 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+// sanitizePEM takes a block of data that should be encoded in PEM and returns only
+// the parts of it that parse as valid PEM blocks.
+func sanitizePEM(data []byte) []byte {
+	var block *pem.Block
+	buf := &bytes.Buffer{}
+	for len(data) > 0 {
+		block, data = pem.Decode(data)
+		if block == nil {
+			return buf.Bytes()
+		}
+		// ignores blocks that can't be serialized
+		_ = pem.Encode(buf, block)
```

Comment:
Open questions:

should we whitelist the PEM block types we allow ("CERTIFICATE", "PRIVATE KEY", "RSA PRIVATE KEY", etc)

When we're validating, are we making sure the data matches the PEM block types? Go's parsing sometimes ignores exact PEM block type matches and will parse any private key from any block type ending in "PRIVATE KEY"

which types does haproxy support?

Is haproxy forgiving about mismatched block type and block content?